utility/associate_nex_unique_ids_with_my_principal_id.go

```diff
@@ -0,0 +1,50 @@
+package utility
+
+import (
+	"github.com/PretendoNetwork/nex-go/v2"
+	"github.com/PretendoNetwork/nex-go/v2/types"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/v2/globals"
+	utility_database "github.com/PretendoNetwork/nex-protocols-common-go/v2/utility/database"
+	utility "github.com/PretendoNetwork/nex-protocols-go/v2/utility"
+	utility_types "github.com/PretendoNetwork/nex-protocols-go/v2/utility/types"
+)
+
+func (commonProtocol *CommonProtocol) associateNexUniqueIDsWithMyPrincipalID(err error, packet nex.PacketInterface, callID uint32, uniqueIDInfos types.List[utility_types.UniqueIDInfo]) (*nex.RMCMessage, *nex.Error) {
+	if err != nil {
+		common_globals.Logger.Error(err.Error())
+		return nil, nex.NewError(nex.ResultCodes.Core.InvalidArgument, "change_error")
+	}
+
+	connection := packet.Sender()
+	endpoint := connection.Endpoint()
+
+	infosLength := len(uniqueIDInfos)
+	if infosLength != 0 {
+		nexError := utility_database.CheckCanAssociateUniqueIDs(commonProtocol.manager, packet.Sender().PID(), uniqueIDInfos)
+		if nexError != nil {
+			common_globals.Logger.Error(nexError.Error())
+			return nil, nexError
+		}
+	}
+
+	nexError := utility_database.ClearPIDUniqueIDAssociations(commonProtocol.manager, packet.Sender().PID())
+	if nexError != nil {
+		common_globals.Logger.Error(nexError.Error())
+		return nil, nexError
+	}
+
+	if infosLength != 0 {
+		nexError = utility_database.UpdateUniqueIDAssociations(commonProtocol.manager, packet.Sender().PID(), uniqueIDInfos, true)
+		if nexError != nil {
+			common_globals.Logger.Error(nexError.Error())
+			return nil, nexError
+		}
+	}
```

Comment:
This can be simplified down to just:


```suggestion
	nexError := utility_database.ClearPIDUniqueIDAssociations(commonProtocol.manager, packet.Sender().PID())
	if nexError != nil {
		common_globals.Logger.Error(nexError.Error())
		return nil, nexError
	}

	nexError = utility_database.UpdateUniqueIDAssociations(commonProtocol.manager, packet.Sender().PID(), uniqueIDInfos, true)
	if nexError != nil {
		common_globals.Logger.Error(nexError.Error())
		return nil, nexError
	}
```

By just checking `CheckCanAssociateUniqueIDs` and the slice length inside `UpdateUniqueIDAssociations`. This also helps reduce the duplicated logic in `associateNexUniqueIDWithMyPrincipalID`